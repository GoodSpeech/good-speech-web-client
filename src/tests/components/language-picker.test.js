import LanguagePicker from '../../components/language-picker';
import { render } from '../test-utils';

const defaultProps = {
  lang: {
    name: 'Spanish',
    code: 'es-AR'
  },
  onChange: () => {}
}

describe('LanguagePicker component', () => {
  beforeEach(() => {
    document.body.innerHtml = '';
  });

  it('Should render default settings', () => {
    const component = render(LanguagePicker, defaultProps);
    expect(component.toJSON()).toMatchSnapshot();  
  });

  it('Should call onChange function when select a language', () => {
    const onChange = jest.fn();
    const component = render(LanguagePicker, defaultProps, {onChange});
    let languagePicker = component.toJSON();

    //Clicking languages button
    languagePicker.children[0].props.onClick({currentTarget: document.body});

    //Language list was open
    expect(document.querySelector('#pick-language')).toBeTruthy();

    //Checks and clicks a language button
    const langButton = document.querySelector('[role="menuitem"]');
    expect(langButton.textContent).toEqual('Afrikaans (Suid-Afrika)');
    langButton.click();

    expect(onChange).toBeCalledWith({
      code: 'af-ZA',
      englishName: 'Afrikaans (South Africa)',
      name: 'Afrikaans (Suid-Afrika)'
    });
  });

});