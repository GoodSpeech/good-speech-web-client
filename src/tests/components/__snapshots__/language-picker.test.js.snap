// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguagePicker component Should render default settings 1`] = `
<span>
  <button
    aria-owns="pick-language"
    className="MuiButtonBase-root-3177716317 MuiButton-root-325725483"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-49836587"
    >
      <span
        className="LanguagePicker-flag-4218349970 flag-icon flag-icon-ar"
      />
      Spanish
      <i
        className="material-icons"
      >
        keyboard_arrow_down
      </i>
    </span>
    <span
      className="MuiTouchRipple-root-3868442396"
    />
  </button>
</span>
`;
